extensions/aggregate_functions.yaml

```diff
@@ -1,18 +1,61 @@
 functions:
-- name: add
-  id: 0
-  arguments:
-    - type: i32
-  variadic: 2..N
-  return: i32
-- name: lt
-  id: 1
-  arguments: 
-    - type: i32
-    - type: i32
-  return: boolean
-- name: and
-  id: 3
-  arguments: - type: boolean
-  variadic: 2..N
-  return: boolean
+  - name: 'sum'
+    nullability: DECLARED_OUTPUT
+    description: "Sum a set of values."
+    options:
+      overflow: [SILENT, SATURATE, ERROR]
+    variants:
+      - name: scalar
+        decomposable: MANY
+        parameters:
+          - K: [i8,i16,i32,i64,fp32,fp64]
+        arguments:
+          - type: K
+        intermediate: K
+        return: K?
+      - name: decimal
+        decomposable: MANY
+        parameters:
+          integer:
+            - P: "1..38"
+            - S: "1..38"
+        arguments:
+          - type: "DECIMAL<P,S>"
+        intermediate: "DECIMAL<38,S>?"
+        return: "DECIMAL<38,S>?"
+  - name: 'avg'
+    nullability: DECLARED_OUTPUT
+    description: "Average a set of values."
+    options:
+      overflow: [SILENT, SATURATE, ERROR]
+    variants:
+      - name: scalar
+        decomposable: MANY
+        parameters:
+          - K: [i8,i16,i32,i64,fp32,fp64]
+        arguments:
+          - type: K?
+        intermediate: "STRUCT<K,i64>"
```

Comment:
Nit: the accumulator probably needs to be the widest type for the particular type class, how would that be expressed (e.g. int16 -> int64 accumulator)?

Reply:
Good point. The intermediate should probably just be STRUCT<i64,i64> for this set of functions. Will post an update.